javaparser-core/src/main/java/com/github/javaparser/ast/comments/CommentsCollection.java

```diff
@@ -67,14 +69,27 @@ public void addComment(Comment comment) {
     }
 
     public boolean contains(Comment comment) {
+        if(!comment.getRange().isPresent()){
+            return false;
+        }
+        Range commentRange = comment.getRange().get();
+        if (commentRange == null) {
+            return false;
+        }
         for (Comment c : getComments()) {
-            // we tolerate a difference of one element in the end column:
-            // it depends how \r and \n are calculated...
-            if (c.getBegin().line == comment.getBegin().line &&
-                    c.getBegin().column == comment.getBegin().column &&
-                    c.getEnd().line == comment.getEnd().line &&
-                    Math.abs(c.getEnd().column - comment.getEnd().column) < 2) {
-                return true;
+            if(!c.getRange().isPresent()){
+                return false;
+            }
+            Range cRange = c.getRange().get();
+            if(cRange!=null) {
+                // we tolerate a difference of one element in the end column:
+                // it depends how \r and \n are calculated...
+                if (cRange.begin.line == commentRange.begin.line &&
```

Comment:
could we replace the first two conditions with `cRange.begin.equals(commentRange.begin)` ?